src/test/java/com/genability/client/api/service/TimeOfUseServiceTests.java

```diff
@@ -213,7 +213,17 @@ public void deletePrivateTouGroupWorksCorrectly() {
 	}
 
 	private void cleanUpPrivateTou(TimeOfUseGroup grp) {
-		touService.deletePrivateTimeOfUseGroup(grp.getLseId(), grp.getTouGroupId());
+		try {
+			// Check if the TOU group has been deleted
+			Response<TimeOfUseGroup> verifyResponse = touService.getTimeOfUseGroup(grp.getLseId(), grp.getTouGroupId());
+			if (verifyResponse.getStatus() == Response.STATUS_SUCCESS) {
+				touService.deletePrivateTimeOfUseGroup(grp.getLseId(), grp.getTouGroupId());
+			}
+		} catch (GenabilityException e) {
+			if (!e.getMessage().contains("404")) {
```

Comment:
You have two calls, which throws the GenabilityException? Wrap the try/catch around just that one -- you generally want try/catches as tightly scoped as possible.

Also explain why a 404-containing message should be ignored (didn't you already check it exists?)